### PR TITLE
Fixes full flow & Adjust local setup for neuro ai

### PR DESCRIPTION
**Pull request by Kozehh**
# 🧠 What is this PR about? 🧠
### Neuro AI
- Adjust local setup to use the same code as the API
- Added local.py file for easy separation between the 2 different setup
- Small adjustments in README & Makefile
- Add type hints 🥰 
- Removed the support for multiple plots because I couldn't make it work 😅
- Fix concentration detection 🧠 

### Backend API
- Adjust Game cycle to be faster
- Fix handleProgress to use the proper payload response from neuro_ai API

Add .env file support to get the PORT # ;)

### CODE REVIEW

File: webapp/vite.config.ts [hunk 1]
```diff
@@ -4,5 +4,8 @@ import react from '@vitejs/plugin-react';
 export default defineConfig({
   plugins: [react()],
   // Patch dev
-  define: process.env.APP_ENV === 'production' ? {} : { global: {} },
-});
+  define: {
+    'process.env.PORT': 5001,
```
Comment (GeorgesMainville) [inline webapp/vite.config.ts:8]: Should be in a .env file instead

Comment (GeorgesMainville) [reply webapp/vite.config.ts:8]: You will need to add `dotenv`. `yarn add dotenv`

**Review by GeorgesMainville — COMMENTED** [review]
.

File: webapp/vite.config.ts (hunk 1 above)
Comment (Kozehh) [reply webapp/vite.config.ts:8]: Ahh lol j'en ai ajouté un, mais j'ai hardcode pareil 😆

File: webapp/vite.config.ts [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
+dotenv.config(); // Load the .env file
 
 export default defineConfig({
   plugins: [react()],
-  // Patch dev
-  define: process.env.APP_ENV === 'production' ? {} : { global: {} },
-});
+  // Patch dev1
+  define: {
+    'process.env.PORT': process.env.PORT || 5001,
```
Comment (GeorgesMainville) [inline webapp/vite.config.ts:11]: We should remove this one too `process.env.PORT`

**Review by GeorgesMainville — COMMENTED** [review]
.

**Pull request merged**